### PR TITLE
Fix disable secretary

### PR DESCRIPTION
**Pull request by x1mx1x**

### CODE REVIEW

File: CharlieBackend.Api/Controllers/SecretariesController.cs [hunk 1]
```diff
@@ -51,13 +51,28 @@ public async Task<ActionResult> PutSecretary(long secretaryId, UpdateSecretaryDt
...
+            {
+                return BadRequest("Unknown secretary id.");
+            }
+
+            var isDisabled = await _accountService.DisableAccountAsync((long)accountId);
+
+            if (isDisabled)
+            {
+                return Ok(await _secretaryService.GetSecretaryByIdAsync(secretaryId));
+            }
```
Comment (sober607) [inline CharlieBackend.Api/Controllers/SecretariesController.cs:73]: Maybe better to have it in service?

**Review by sober607 — APPROVED** [review]
2 tasks is 1 pull request.
Good

**Review by SerjKocherga — APPROVED** [review]

**Review by SnipeFrost — APPROVED** [review]

File: CharlieBackend.Business/Services/FileImportService.cs [hunk 2]
```diff
@@ -42,6 +42,8 @@ public async Task<Result<List<StudentGroupFileModel>>> ImportFileAsync(ImportFil
             {
                 if (property.Name != Convert.ToString(wsGroups.Cell($"{charPointer}1").Value))
                 {
+
+
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/FileImportService.cs:46]: Remove additional newlines

File: CharlieBackend.Business/Services/FileImportService.cs [hunk 3]
```diff
@@ -88,6 +90,7 @@ public async Task<Result<List<StudentGroupFileModel>>> ImportFileAsync(ImportFil
                 {
                     _unitOfWork.Rollback();
 
+
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/FileImportService.cs:93]: Remove additional newline

File: CharlieBackend.Business/Services/FileImportService.cs [hunk 4]
```diff
@@ -96,6 +99,7 @@ public async Task<Result<List<StudentGroupFileModel>>> ImportFileAsync(ImportFil
             await _unitOfWork.CommitAsync();
 
             return Result<List<StudentGroupFileModel>>
+
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/FileImportService.cs:102]: Remove additional newline

File: CharlieBackend.Business/Services/LessonService.cs [hunk 5]
```diff
@@ -70,21 +71,21 @@ public async Task<IList<LessonDto>> GetAllLessonsAsync()
...
-        public async Task<Lesson> AssignMentorToLessonAsync(AssignMentorToLessonDto ids)
+        public async Task<Result<Lesson>> AssignMentorToLessonAsync(AssignMentorToLessonDto ids)
         {
-            var mentorToAssign = await _unitOfWork.MentorRepository.GetMentorByAccountIdAsync(ids.MentorId);
+            var mentorToAssign = await _unitOfWork.MentorRepository.GetMentorByIdAsync(ids.MentorId);
 
             if (mentorToAssign == null)
             {
-                throw new NullReferenceException();
+                return Result<Lesson>.Error(ErrorCode.NotFound, "this mentor id doesn't exist");
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/LessonService.cs:80]: Indicate which id does not exist in error message

**Review by whereismyroot — CHANGES_REQUESTED** [review]
Added small comments.
Please merge dev into this branch and fix build error

File: CharlieBackend.Api/Controllers/SecretariesController.cs (hunk 1 above)
Comment (Vlad1023) [reply CharlieBackend.Api/Controllers/SecretariesController.cs:73]: Better move this logic to service and use Result T

**Review by Vlad1023 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**